File: 0442-find-all-duplicates-in-an-array/0442-find-all-duplicates-in-an-array.cpp
class Solution {
public:
    vector<int> findDuplicates(vector<int>& nums) {
        vector<int>dupl;
        for(int i=0;i<nums.size();i++){
          if(nums[abs(nums[i])-1] < 0){
           dupl.push_back(abs(nums[i]));
          }
            nums[abs(nums[i])-1] = nums[abs(nums[i])-1]*(-1);

        }
        return dupl;
    }
};

File: 0442-find-all-duplicates-in-an-array/0442-find-all-duplicates-in-an-array.java
class Solution {
    public List<Integer> findDuplicates(int[] nums) {
        List<Integer>ans = new ArrayList<>();
        for(int i=0;i<nums.length;i++){
           int idx = Math.abs(nums[i]);

           if(nums[idx-1] < 0){
            ans.add(Math.abs(nums[i]));
           }
           else{
            nums[idx -1] = -nums[idx-1];
           }
        }

        return ans;
    }
}